paddle/fluid/operators/fused/fused_attention_op.cc

```diff
@@ -157,17 +197,32 @@ class FusedAttentionOp : public framework::OperatorWithKernel {
       ctx->SetOutputDim("Ln2Variance", {x_dim[0] * x_dim[1]});
       ctx->SetOutputDim("BiasDropoutResidualOut", ctx->GetInputDim("X"));
     }
-    // [batch_size, seq_len, 3, num_head, head_size]
-    ctx->SetOutputDim("QKVOut",
-                      {x_dim[0], x_dim[1], y_dim[0], y_dim[1], y_dim[2]});
 
-    if (ctx->HasInput("QKVBias")) {
-      ctx->SetOutputDim("QKVBiasOut",
+    if (!transpose_qkv_wb) {
+      // [batch_size, seq_len, 3, num_head, head_size]
+      ctx->SetOutputDim("QKVOut",
                         {x_dim[0], x_dim[1], y_dim[0], y_dim[1], y_dim[2]});
+
+      if (ctx->HasInput("QKVBias")) {
+        ctx->SetOutputDim("QKVBiasOut",
+                          {x_dim[0], x_dim[1], y_dim[0], y_dim[1], y_dim[2]});
+      }
+      // [3, batch_size, num_head, seq_len, head_size]
+      ctx->SetOutputDim("TransposeOut2",
+                        {y_dim[0], x_dim[0], y_dim[1], x_dim[1], y_dim[2]});
+    } else {
+      // [batch_size, seq_len, 3 * hidden_size]
+      ctx->SetOutputDim("QKVOut",
+                        {x_dim[0], x_dim[1], 3 * num_heads * dim_head});
```

Comment:
`{x_dim[0], x_dim[1], 3, num_heads, dim_head}`?

Reply:
Although the shape `{x_dim[0], x_dim[1], 3 * hidden_size}` and the shape `{x_dim[0], x_dim[1], 3, num_heads, dim_head}` are same for cublas gemm. I prefer set the shape as I coded and add some `Resize` operations in the kernel. Mainly for maintenance purpose. When we enable `transpose_qkv_w`, the qkv_w's shape is `[h, 3h]` and the input x's shape is `[b, s, h]`, so set the shape of output of qkv gemm as `{x_dim[0], x_dim[1], 3 * hidden_size}` is more straight forward for the maintainers. I think it will introduce some codes reading obstacles if we set the shape of qkv output as {x_dim[0], x_dim[1], 3, num_heads, dim_head}.

Reply:
Follow these comment and refine the code, to make the shape of the tensor is consistent before and after the tensor.